src/VisualStudio/Core/Def/Implementation/Preview/PreviewEngine.cs

```diff
@@ -194,10 +194,10 @@ public int GetWarning(out string pbstrWarning, out int ppcwlWarningLevel)
         public void UpdatePreview(DocumentId documentId, SpanChange spanSource)
         {
             var updatedSolution = _topLevelChange.GetUpdatedSolution(applyingChanges: false);
-            var document = updatedSolution.GetDocument(documentId);
+            var document = updatedSolution.GetDocument(documentId) ?? updatedSolution.GetAdditionalDocument(documentId);
             if (document != null)
             {
-                _updater.UpdateView(document, spanSource);
+                _updater.UpdateView(document, spanSource, isAdditionalDocument: !(document is Document));
```

Comment:
is this flag needed? UpdateView should be able to check it itself since document is given.

also can we have extension method that is
 GetTextDocument(this solution, documentid) { 
return solution.GetDocument(documentId) ?? solution.GetAdditionalDocument(documentId) 
}

looks like useful one to have.